### DISCUSSION TITLE
OOP2 support for a new Libre 2 RU

### DISCUSSION THREAD

**Discussion by plastik21** [NightscoutFoundation/xDrip#3523] [General]
Please, somebody give me a contact to OOP2 developers. I'm gonna prey them to support Russian Libre 2 :)

**Comment by Navid200** [comment]
@PhilTU 
Is this a different device than the one that has an open PR?

**Reply by PhilTU** [reply]
Yes. https://github.com/NightscoutFoundation/xDrip/pull/3514 addresses "UK Libre 2 plus" sensor with patchinfo c6-09-31 ID
This was a case of a new serial number, with the same structure as before (and one more day life).

The RU sensor here and at https://github.com/NightscoutFoundation/xDrip/discussions/3439#discussioncomment-9253213 has patchinfo 2b-0a-39-08 and according to @gui-dos is a Generation 2 device along the lines of the US Libre 2 which needs a different NFC command to access it (his comments seem to have disappeared from the discussion).

extension SensorType {
    init(patchInfo: PatchInfo) {
        self = switch patchInfo[0] {
        case 0xDF, 0xA2: .libre1
        case 0xE5, 0xE6: .libreUS14day
        case 0x70:       .libreProH
        case 0x9D, 0xC5: .libre2
        case 0x76, 0x2B:
            patchInfo[3] == 2 ? .libre2US :
            patchInfo[3] == 4 ? .libre2CA :
            patchInfo[3] == 8 ? .libre2RU :
            patchInfo[2] >> 4 == 7 ? .libreSense :
                .unknown
        case 0x2C:       .libre2Plus

**Reply by yafoxins** [reply]
Any word on if they'll be getting them in xDrip? @jamorham
